bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,291 @@
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = requireNonNull(httpClientFactory, "httpClientFactory");
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+            updateStatus(ONLINE);
+        } catch (Exception ex) {
+            logger.error("Cannot start server!", ex);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Cannot start server! " + ex.getMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        loadConfigs();
+        var missingUsername = "".equals(username);
+        var missingPassword = password == null || password.length == 0;
+        var missingUrl = "".equals(url);
+        if (missingUsername || missingPassword || missingUrl) {
+            var sb = new StringBuilder();
+            sb.append("Missing configuration!\n");
+            sb.append(missingUsername ? "❌" : "✅").append(" username\n");
+            sb.append(missingPassword ? "❌" : "✅").append(" password\n");
+            sb.append(missingUrl ? "❌" : "✅").append(" url\n");
+            sb.append("Please check your configuration!\n");
+            logger.error("{}", sb);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, sb.toString());
+            return;
+        }
+        var httpClient = new JettyHttpClient(httpClientFactory.getCommonHttpClient());
+        var x = salusApi = new SalusApi(username, password, url, httpClient, GsonMapper.INSTANCE);
+        logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName() + "[" + username.replace(".", "_") + "]");
+        try {
+            x.findDevices();
+        } catch (Exception ex) {
+            var msg = "Cannot connect to Salus Cloud! Probably username/password mismatch!";
+            logger.error(msg, ex);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, msg + " " + ex.getMessage());
+            return;
+        }
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
+                .expireAfterWrite(Duration.ofSeconds(propertiesRefreshInterval))
+                .refreshAfterWrite(Duration.ofSeconds(propertiesRefreshInterval)).build(this::loadPropertiesForDevice);
+        var scheduledPool = ThreadPoolManager.getScheduledPool("Salus");
+        this.scheduledFuture = scheduledPool.scheduleWithFixedDelay(this::refreshCloudDevices, refreshInterval * 2,
+                refreshInterval, SECONDS);
+
+        // done
+        updateStatus(ONLINE);
+    }
+
+    private void loadConfigs() {
+        var config = this.getConfig();
+        username = (String) config.get("username");
+        password = ((String) config.get("password")).toCharArray();
+        url = (String) config.get("url");
+        if ("".equals(url)) {
+            url = "https://eu.salusconnect.io";
```

Comment:
See earlier comment about seperate config class, That would also make it easier tot set default values, lik with this url.